### DISCUSSION TITLE
Couchdb3.1.1: Logout not working

### DISCUSSION THREAD

**Discussion by RJNaamthos** [apache/couchdb#3246] [Q&A]
[NOTE]: # ( ^^ Provide a general summary of the issue in the title above. ^^ )

## Description
Enabled "require_valid_user" attribute under [chttpd] section to true and modified authentication_handlers value to only default_authentication_handler

[chttpd]
require_valid_user = true
authentication_handlers = {chttpd_auth, default_authentication_handler}

[chttpd_auth]
authentication_db = _users

when accessing CouchdbUI fauxton, browser pop up comes for authentication. Once providing the credentials, able to login and access couchdb UI.  However on click of logout button, nothing happens. It does not logout the User.

[NOTE]: # ( Describe the problem you're encountering. )
[TIP]:  # ( Do NOT give us access or passwords to your actual CouchDB! )

## Steps to Reproduce
- Modified the values in local.ini.

[chttpd]
require_valid_user = true
authentication_handlers = {chttpd_auth, default_authentication_handler}

[chttpd_auth]
authentication_db = _users

- Login through CouchdbUI fauxton, browser pop up comes for authentication. Provide credentials.
- Once login through credentials, now click on logout.
- Logout does not work.

[NOTE]: # ( Include commands to reproduce, if possible. curl is preferred. )

## Expected Behaviour
- User should be logged out on click of logout.

[NOTE]: # ( Tell us what you expected to happen. )

## Your Environment

[TIP]:  # ( Include as many relevant details about your environment as possible. )
[TIP]:  # ( You can paste the output of curl http://YOUR-COUCHDB:5984/ here. )

* CouchDB version used: Couchdb 3.1.1
* Browser name and version:  Firefox, Chrome
* Operating system and version: CentOS 7.8

## Additional Context

[TIP]:  # ( Add any other context about the problem here. )

**Comment by ustun** [comment]
> browser pop up comes for authentication

@RJNaamthos  this is browser's http authentication, so browser is caching the login. it is tricky to get logout working robustly there.. see https://www.php.net/manual/en/features.http-auth.php for some techniques to force logout in some browsers. I don't know if there is anything in couchdb to combat that (I'm not a couchdb dev, just another user)

It should log out if you close the browser tab.

Does logout work if you login from the normal web form? I always use that because of this issue.

**Reply by RJNaamthos** [reply]
Couchdb1.6.1/1.7.1 had mechanism where logout works properly when login through browser pop up. The below properties were used in 1.6.1/1.7.1

[couch_httpd_auth]
require_valid_user = true

[httpd]
authentication_handlers = {couch_httpd_auth, default_authentication_handler}

Similar behavior(log out) for Couchdb 3.1.1 should also work as only change which is required is to add these above properties under [chttpd].

**Comment by RJNaamthos** [comment]
Can anyone look into this and provide some suggestions. Thanks.

**Reply by ustun** [reply]
Seems like a duplicate of https://github.com/apache/couchdb-fauxton/issues/1286 No suggestions there either though.

**Comment by RJNaamthos** [comment]
Can someone look into this and provide some suggestions. Thanks.
